platform/keyring.impl/src/org/netbeans/modules/keyring/gnome/libsecret/GnomeLibSecretProvider.java

```diff
@@ -0,0 +1,138 @@
+    private static final String KEY = "key"; // NOI18N
+
+    private final String appName;
+    private final LibSecret.SecretSchema schema;
+
+    public GnomeLibSecretProvider() {
+        appName = getAppName();
+
+        schema = new SecretSchema();
+        schema.name = appName;
+        schema.flags = LibSecret.SECRET_SCHEMA_NONE;
+        schema.attributes[0] = new SecretSchemaAttribute();
+        schema.attributes[0].name = KEY;
+        schema.attributes[0].type = LibSecret.SECRET_SCHEMA_ATTRIBUTE_STRING;
+    }
+
+    private String getAppName() {
+        try {
+            return MessageFormat.format(
+                    NbBundle.getBundle("org.netbeans.core.windows.view.ui.Bundle").getString("CTL_MainWindow_Title_No_Project"),
+                    "…");
+        } catch (MissingResourceException x) {
+            return "NetBeans"; // NOI18N
+        }
+    }
+
+    @Override
+    public boolean enabled() {
+        if (Boolean.getBoolean("netbeans.keyring.no.native")) {
+            LOG.fine("native keyring integration disabled");
+            return false;
+        }
+
+        try {
+            read("NoNeXiStEnT"); // NOI18N
+            return true;
+        } catch (RuntimeException e) {
+            LOG.log(Level.FINE, null, e);
+            return false;
+        }
+    }
+
+    @Override
+    public char[] read(String key) {
+        PointerByReference gerrorBuffer = new PointerByReference();
+
+        Pointer pointer = LibSecret.INSTANCE.secret_password_lookup_sync(schema, null, gerrorBuffer, KEY, key);
+
+        if (gerrorBuffer.getValue() != null) {
+            processError(gerrorBuffer);
+            return null;
+        }
+
+        if (pointer == null) {
+            return null;
+        }
+
+        String secret = pointer.getString(0);
+        if (secret == null) {
+            LOG.warning("Secret is null");
+            return null;
+        }
+
+        return secret.toCharArray();
+    }
+
+    @Override
+    public void save(String key, char[] password, String description) {
+        PointerByReference gerrorBuffer = new PointerByReference();
+
+        String label = appName + " - " + (description != null ? description : key);
+        LibSecret.INSTANCE.secret_password_store_sync(schema, LibSecret.SECRET_COLLECTION_DEFAULT, label, new String(password), null, gerrorBuffer, KEY, key);
+
+        if (gerrorBuffer.getValue() != null) {
+            processError(gerrorBuffer);
+        }
+    }
+
+    @Override
+    public void delete(String key) {
+        PointerByReference gerrorBuffer = new PointerByReference();
+
+        LibSecret.INSTANCE.secret_password_clear_sync(schema, null, gerrorBuffer, KEY, key);
+
+        if (gerrorBuffer.getValue() != null) {
+            processError(gerrorBuffer);
+        }
+    }
+
+    private void processError(PointerByReference gerrorBuffer) throws IllegalArgumentException {
```

Comment:
the old GnomeProvider also does not throw exceptions when the native code returns an error, just returns null in case of `read` and just logs the error in case of `read`, `save` and `delete`. 
It throws an exception when the native library is not available which is used in the `enabled` method.